app/src/lib/stores/pull-request-store.ts

```diff
@@ -48,25 +50,48 @@ export class PullRequestStore {
 
       await this.writePRs(raw, repository)
 
-      const results = await this.getPullRequests(repository)
-
-      await this.getStatusForPRs(results, repository, account)
-
-      this.pullRequests = await this.getPullRequests(repository)
-      this.emitUpdate()
+      const prs = await this.getPullRequests(repository)
+      await this.refreshStatusForPRs(prs, repository, account)
     } catch (error) {
       log.warn(`Error refreshing pull requests for '${repository.name}'`, error)
       this.emitError(error)
+    } finally {
+      this.changeActiveFetchCount(repository, c => c - 1)
+      this.emitUpdate(repository)
```

Comment:
Is there ever a time we don't want to emit after calling `changeActiveFetchCount`? If not, I think we should move `this.emitUpdate` into `changeActiveFetchCount`.

Reply:
Yeah I went back and forth on that. I'm into putting the `emitUpdate` in there 👍